File: 01-Backend/DatajudClient.API/DatajudClient.Domain/Models/Processos/Processo.cs
using DatajudClient.Domain.Models.Endereco;
using DatajudClient.Domain.Models.Tribunais;

namespace DatajudClient.Domain.Models.Processos
{
    public class Processo : ModelBase
    {
        public string NumeroProcesso { get; set; }
        public string NomeCaso{ get; set; }
        public string Vara { get; set; }
        public string Comarca { get; set; }
        public string Observacao { get; set; }
        public DateTime? UltimoAndamento { get; set; }
        public DateTime? UltimaAtualizacao { get; set; }
        public int EstadoId { get; set; }
        public virtual Estado Estado { get; set; }
        public int TribunalId { get; set; }
        public virtual Tribunal Tribunal { get; set; }
        public virtual List<AndamentoProcesso>? Andamentos { get; set; }

        public Processo()
        {
            NumeroProcesso = string.Empty;
            NomeCaso = string.Empty;
            Vara = string.Empty;
            Comarca = string.Empty;
            Observacao = string.Empty;
        }

        /// <summary>
        /// Retorna o número do processo formatado. Caso o número do processo não tenha 20 caracteres, retorna o número do processo sem formatação.
        /// </summary>
        /// <returns></returns>
        public string GetNumeroFormatado()
        {
            if(NumeroProcesso.Length == 20)
            {
                return $"{NumeroProcesso.Substring(0, 7)}.{NumeroProcesso.Substring(7, 2)}.{NumeroProcesso.Substring(9, 4)}.{NumeroProcesso.Substring(13, 4)}.{NumeroProcesso.Substring(17, 2)}";
            }
            else
            {
                return NumeroProcesso;
            }
        }
    }
}

File: 02-Frontend/DatajudClient.Front/src/app/models/processo/processo.ts
import { IProcesso } from "../../interfaces/processo/processo";
import { Estado } from "../../interfaces/endereco/estado";
import { Tribunal } from "../../interfaces/tribunal/tribunal";
import { AndamentoProcesso } from "../../interfaces/processo/andamentoProcesso";
import { StringTools } from "../../util/stringtools";

export class Processo implements IProcesso {
    id: number;
    numeroProcesso: string;
    nomeCaso: string;
    vara: string | null;
    comarca: string;
    observacao: string | null;
    ultimoAndamento: Date;
    ultimoAndamentoDescricao: string;
    ultimaAtualizacao: Date;
    estado: Estado;
    tribunal: Tribunal;
    Andamentos: AndamentoProcesso[] | null;

    constructor() {
      this.id = 0;
      this.numeroProcesso = '';
      this.nomeCaso = '';
      this.vara = '';
      this.comarca = '';
      this.observacao = '';
      this.ultimoAndamento = new Date();
      this.ultimoAndamentoDescricao = '';
      this.ultimaAtualizacao = new Date();
      this.estado = {} as Estado;
      this.tribunal = {} as Tribunal;
      this.Andamentos = [];
    }

    getNumeroProcessoFormatado(): string {
        return StringTools.apenasNumeros(this.numeroProcesso).replace(/(\d{7})(\d{2})(\d{4})(\d{1})(\d{2})(\d{4})/, "$1-$2.$3.$4.$5.$6");
    }
}
